### ISSUE TITLE
changing .constraints-container to display:flex has possibly undesirable side-effects

### ISSUE THREAD

**Issue by jrochkind** [projectblacklight/blacklight#2552]
#2469 changes the `.constraints-container` to `display: flex` in order to improve pixel alignment. 

But for long lines, it can change the layout in other ways that may be undesirable. 

## After #2469, with `display:flex`, a stock BL 7.21.2 app:

![Screen Shot 2021-11-23 at 4 32 36 PM](https://user-images.githubusercontent.com/149304/143132699-7d82c6b5-c4b3-4cd0-ae67-422014be04fc.png)

![Screen Shot 2021-11-23 at 4 33 07 PM](https://user-images.githubusercontent.com/149304/143132698-394dd875-4e28-4023-8d2b-6c219469ab76.png)

## Manually changing `.constraints-container` back to `display:block`

![Screen Shot 2021-11-23 at 4 35 52 PM](https://user-images.githubusercontent.com/149304/143132951-dbfd43b7-09fa-475b-9437-6a7e79454150.png)

![Screen Shot 2021-11-23 at 4 33 39 PM](https://user-images.githubusercontent.com/149304/143132771-0f14f277-b802-418e-bb67-a1efd783c8a5.png)

Some issues with lack of spacing between the line when it wraps, but a lot closer to desirable I think?  

(Also note the `1-2 of 2` seems mis-aligned in both examples, but that's a separate issue I think)

## solution?

I wonder if the `display: flex` should be reverted, and another solution found to the pixel alignment? Alternately, I guess there might be a way to add more CSS to make it wrap better even under display:flex (maybe also need to add a `flex-wrap: wrap`, `white-space: no-wrap`, and possibly other flex alignment properties?), but just want to make sure we're not adding workarounds on top of workarounds in increasingly hairy ways (I don't personally understand the original issue in #2469 or why `display:flex` solved it). 

Any thoughts @cbeer?
